lib/shopify_theme/cli.rb

```diff
@@ -36,18 +36,28 @@ class Cli < Thor
     end
 
     desc "check", "check configuration"
-    def check
-      if ShopifyTheme.check_config
-        say("Configuration [OK]", :green)
+    def check(exit_on_failure=false)
+      result = APIChecker.new(ShopifyTheme).test_connectivity
+      if result.api_down?
+        say("Cannot connect to Shopify. API appears to be down", :red)
+        say("Visit http://status.shopify.com for more details", :yello)
+      elsif result.invalid_config?
+        say("Cannot connect to Shopify. Configuration is invalid.", :red)
+        say("Verify that your API Key, Password and Domain are correct.", :yellow)
+        say("Visit https://github.com/shopify/shopify_theme#configuration for more details", :yellow)
+        say("If your shop domain is correct, the following URL should take you to the Private Apps page for the shop:", :yellow)
+        say("  https://#{config[:store]}/admin/apps/private", :yellow)
       else
-        say("Configuration [FAIL]", :red)
+        say("Shopify API is accessible and Configuration is valid", :green) unless exit_on_failure
       end
+      exit(1) if result.cannot_access_api? && exit_on_failure
     end
 
     desc "configure API_KEY PASSWORD STORE THEME_ID", "generate a config file for the store to connect to"
     def configure(api_key=nil, password=nil, store=nil, theme_id=nil)
       config = {:api_key => api_key, :password => password, :store => store, :theme_id => theme_id}
       create_file('config.yml', config.to_yaml)
+      check(true)
```

Comment:
If the check fails, do we want to leave the config file in place with invalid values?